lib/src/supabase_auth.dart

```diff
@@ -300,12 +300,101 @@ extension GoTrueClientSignInProvider on GoTrueClient {
       scopes: scopes,
       queryParams: queryParams,
     );
-    final url = Uri.parse(res.url!);
-    final result = await launchUrl(
-      url,
-      mode: authScreenLaunchMode,
-      webOnlyWindowName: '_self',
+    final uri = Uri.parse(res.url!);
+
+    bool? result;
+    if (authScreenLaunchMode == LaunchMode.inAppWebView) {
+      result = await showModalBottomSheet<bool>(
+        context: context,
+        isScrollControlled: true,
+        builder: (context) => _OAuthSignInWebView(
+          uri: uri,
+          redirectTo: redirectTo,
+        ),
+      );
+    } else {
+      result = await launchUrl(
+        uri,
+        mode: authScreenLaunchMode,
+        webOnlyWindowName: '_self',
+      );
+    }
+    return result ?? false;
+  }
+}
+
+class _OAuthSignInWebView extends StatefulWidget {
+  const _OAuthSignInWebView({
+    Key? key,
+    required this.uri,
```

Comment:
Nothing critical, but personally I like a more descriptive variable name, maybe something like this:
```suggestion
    required this.oAuthUri,
```